homeassistant/components/emulated_hue/config.py

```diff
@@ -0,0 +1,214 @@
+        # don't really have a concept of being off
+        off_maps_to_on_domains = conf.get(CONF_OFF_MAPS_TO_ON_DOMAINS)
+        if isinstance(off_maps_to_on_domains, list):
+            self.off_maps_to_on_domains = set(off_maps_to_on_domains)
+        else:
+            self.off_maps_to_on_domains = DEFAULT_OFF_MAPS_TO_ON_DOMAINS
+
+        # Get whether or not entities should be exposed by default, or if only
+        # explicitly marked ones will be exposed
+        self.expose_by_default = conf.get(
+            CONF_EXPOSE_BY_DEFAULT, DEFAULT_EXPOSE_BY_DEFAULT
+        )
+
+        # Get domains that are exposed by default when expose_by_default is
+        # True
+        self.exposed_domains = set(
+            conf.get(CONF_EXPOSED_DOMAINS, DEFAULT_EXPOSED_DOMAINS)
+        )
+
+        # Calculated effective advertised IP and port for network isolation
+        self.advertise_ip = conf.get(CONF_ADVERTISE_IP) or self.host_ip_addr
+
+        self.advertise_port = conf.get(CONF_ADVERTISE_PORT) or self.listen_port
+
+        self.entities = conf.get(CONF_ENTITIES, {})
+
+        self._entities_with_hidden_attr_in_config = {}
+        for entity_id in self.entities:
+            hidden_value = self.entities[entity_id].get(CONF_ENTITY_HIDDEN)
+            if hidden_value is not None:
+                self._entities_with_hidden_attr_in_config[entity_id] = hidden_value
+
+        # Get whether all non-dimmable lights should be reported as dimmable
+        # for compatibility with older installations.
+        self.lights_all_dimmable = conf.get(CONF_LIGHTS_ALL_DIMMABLE)
+
+    async def async_setup(self) -> None:
+        """Set up and migrate to storage."""
+        self.store = storage.Store(self.hass, DATA_VERSION, DATA_KEY)  # type: ignore[arg-type]
+        self.numbers = (
+            await storage.async_migrator(
+                self.hass, self.hass.config.path(NUMBERS_FILE), self.store
+            )
+            or {}
+        )
+
+    def entity_id_to_number(self, entity_id: str) -> str:
+        """Get a unique number for the entity id."""
+        if self.type == TYPE_ALEXA:
+            return entity_id
+
+        # Google Home
+        for number, ent_id in self.numbers.items():
+            if entity_id == ent_id:
+                return number
+
+        number = "1"
+        if self.numbers:
+            number = str(max(int(k) for k in self.numbers) + 1)
+        self.numbers[number] = entity_id
+        assert self.store is not None
+        self.store.async_delay_save(lambda: self.numbers, SAVE_DELAY)
+        return number
+
+    def number_to_entity_id(self, number: str) -> str | None:
+        """Convert unique number to entity id."""
+        if self.type == TYPE_ALEXA:
+            return number
+
+        # Google Home
+        assert isinstance(number, str)
+        return self.numbers.get(number)
+
+    def get_entity_name(self, entity: State) -> str:
```

Comment:
I'd rename the parameter to `state`.

Reply:
I agree, I kept the names when I relocated this code.  There are 5-6 places that it should be renamed.  I plan on doing that in the next PR since I didn't want to make this any larger.

Reply:
I left this as-is for the next pass as changing all the places made this PR much larger.

Reply:
https://github.com/home-assistant/core/pull/73004